src/utility.jl

```diff
@@ -74,3 +75,260 @@ function visualize_flow(method::ColorBased, flow::Array{SVector{2, Float64}, 2};
+end
+
+
+"""
+```
+evaluate_flow_error(ground_truth_flow, estimated_flow, EndpointError())
+```
+
+Returns a 2-Dimensional array that matches the dimensions of the input flow vector and
+that depicts the end point error between the estimated flow and the ground truth flow.
+
+# Details
+If the estimated flow at a point is (u0, v0) and ground truth flow is (u1, v1), then 
+error will be sqrt[(u0 - u1)^2 + (v0 - v1)^2] at that point.
+
+# Arguments
+The flow parameters needs to be two-dimensional arrays of length-2 vectors (of type SVector) 
+which represent the displacement of each pixel.
+
+# Example
+
+Compute the end point error between two flows.
+```julia
+
+using ImageTracking
+
+result = evaluate_flow_error(ground_truth_flow, estimated_flow, EndpointError())
+
+imshow(result)
+```
+
+# References
+[1] S. Baker, D. Scharstein, JP Lewis, S. Roth, M.J. Black, and R. Szeliski. A database and
+evaluation methodology for optical flow.International Journal of Computer Vision, 92(1):1–31, 2011.
+"""
+function evaluate_flow_error(ground_truth_flow::Array{SVector{2, Float64}, 2}, estimated_flow::Array{SVector{2, Float64}, 2}, error_type::EndpointError)
+    result = Array{Float64, 2}(undef, size(estimated_flow))
+    for i in 1:size(estimated_flow)[1]
+        for j in 1:size(estimated_flow)[2]
+            p1 = estimated_flow[i, j]
+            p2 = ground_truth_flow[i, j]
+
+            if is_flow_known(p1) && is_flow_known(p2) 
+                δ = p1 - p2
+                result[i, j] = sqrt(sum(δ.^2))
+            else
+                result[i, j] = NaN
+            end
+        end
+    end
+    
+    return result
+end
+
+
+"""
+```
+evaluate_flow_error(ground_truth_flow, estimated_flow, AngularError())
+```
+
+Returns a 2-Dimensional array that matches the dimensions of the input flow vector and
+that depicts the angle error between the estimated flow and the ground truth flow.
+
+# Details
+If the estimated flow at a point is (u0, v0) and ground truth flow is (u1, v1), it 
+calculates the angle between (u0, v0, 1) and (u1, v1, 1) vectors as measure for error. 
+
+# Arguments
+The flow parameters needs to be two-dimensional arrays of length-2 vectors (of type SVector) 
+which represent the displacement of each pixel.
+
+# Example
+
+Compute the angle error between two flows.
+```julia
+
+using ImageTracking
+
+result = evaluate_flow_error(ground_truth_flow, estimated_flow, AngularError())
+
+imshow(result)
+```
+
+# References
+[1] S. Baker, D. Scharstein, JP Lewis, S. Roth, M.J. Black, and R. Szeliski. A database and
+evaluation methodology for optical flow.International Journal of Computer Vision, 92(1):1–31, 2011.
+"""
+function evaluate_flow_error(ground_truth_flow::Array{SVector{2, Float64}, 2}, estimated_flow::Array{SVector{2, Float64}, 2}, error_type::AngularError)
+    result = Array{Float64, 2}(undef, size(estimated_flow))
+    for i in 1:size(estimated_flow)[1]
+        for j in 1:size(estimated_flow)[2]
```

Comment:
I suggest you use `CartesianIndices` as per my comment in the other function.
Then do 
```julia
p1 = ground_truth_flow[i]
p2 = estimated_flow[i]
if is_flow_know(p1) && is_flow_know(p2)
  p1 = push(p1, 1) # No need for the append! convert etc.
  p2 = push(p2, 1)
  # Etc.
```

Reply:
Done